src/crypto-providers/util.ts

```diff
@@ -273,6 +276,24 @@ const ipv6ToString = (ipv6: Buffer | undefined): string | undefined => {
 
 const rewardAddressToPubKeyHash = (address: Buffer) => address.slice(1)
 
+const deviceVersionToStr = (
+  deviceVersion: DeviceVersion,
+): string => `${deviceVersion.major}.${deviceVersion.minor}.${deviceVersion.patch}`
+
+const isDeviceVersionGTE = (
+  deviceVersion: DeviceVersion,
+  thresholdVersion: DeviceVersion,
+): boolean => deviceVersion.major > thresholdVersion.major
+  || (
+    deviceVersion.major === thresholdVersion.major
+    && deviceVersion.minor > thresholdVersion.minor
+  )
+  || (
+    deviceVersion.major === thresholdVersion.major
+    && deviceVersion.minor === thresholdVersion.minor
+    && deviceVersion.patch >= thresholdVersion.patch
+  )
+
 export {
   isShelleyPath,
   isStakingPath,
```

Comment:
isStakingPath check seems incomplete: it should end with /2/0, not just /2/whatever.
And length is not checked for paths.

Reply:
Also, I have found no check for coin 1815 in paths anywhere.

Reply:
Regarding the `isStakingPath`, previously it was considered the delegation to multiple pools could be done by changing the last index of the staking path, and also, we necessarily don't enforce the user to create staking paths with 0 in the last index. In theory, someone could create an address with staking from `1815'/1852'/0'/2/1`. 

Thanks for bringing attention to this, I checked how we validate path as arguments, and the `isBIP32Path` only checks whether the path is composed of numbers. We should rename it to `isCardanoPath` or something similar, check for 1815, 1852, and make sure the length doesn't exceed the length hw wallets support. I ll look into it or make it into an issue so we definitely address this.

Reply:
Created issue for this, will address in another PR.